d3a_interface/dataclasses.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+
+
+@dataclass
```

Comment:
I like the dataclass usage here, however I think it is worth it to try it with Pypy 3.7 to validate that no issues occur (because the dataclasses were introduced in Python 3.7, would be great to check the Pypy implementation to rule out any incompatibilities)

Reply:
AFAIU it works (couldn't validate this as pyenv is failing)

Reply:
> This backport assumes that dict objects retain their insertion order. This is true in the language spec for Python 3.7 and greater. Since this is a backport to Python 3.6, it raises an interesting question: does that guarantee apply to 3.6? For CPython 3.6 it does. As of the time of this writing, it’s also true for all other Python implementations that claim to be 3.6 compatible, of which there are none. Any new 3.6 implementations are expected to have ordered dicts
This is from the pypi dataclasses documentation [link](https://pypi.org/project/dataclasses/)

Reply:
This link is valid only for the backport, therefore it does not really say much about Pypy because we do not use the backport anyways (we do not run `pip install dataclass`). I strongly recommend creating a local d3a virtualenv with Pypy in order to test such cases (even without Pyenv you can install pypy3 directly in your machine and use it instead). For now feel free to merge this, since I am also 99% sure that this will work, I am mentioning the above in order to be prepared for any future relevant issues.